internal/services/springcloud/spring_cloud_configuration_service_resource.go

```diff
@@ -0,0 +1,316 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceSpringCloudConfigurationService() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSpringCloudConfigurationServiceCreateUpdate,
+		Read:   resourceSpringCloudConfigurationServiceRead,
+		Update: resourceSpringCloudConfigurationServiceCreateUpdate,
+		Delete: resourceSpringCloudConfigurationServiceDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.SpringCloudConfigurationServiceID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"default",
+				}, false),
+			},
+
+			"spring_cloud_service_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SpringCloudServiceID,
+			},
+
+			"repository": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+
+						"label": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+
+						"patterns": {
+							Type:     pluginsdk.TypeSet,
+							Required: true,
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+							},
+						},
+
+						"uri": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+
+						"host_key": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+
+						"host_key_algorithm": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+
+						"password": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+
+						"private_key": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+
+						"search_paths": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+							},
+						},
+
+						"strict_host_key_checking": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"username": {
```

Comment:
should this just be
```suggestion
						"user": {
```

Reply:
I think maybe not necessary. Because this aligns with https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/spring_cloud_service#username